server/bootstrap/src/org/labkey/filters/ContentSecurityPolicyFilter.java

```diff
@@ -166,4 +194,9 @@ public static String getScriptNonceHeader(HttpServletRequest request)
     }
 
     private static final SecureRandom rand = new SecureRandom();
-}
\ No newline at end of file
+
+    public static void registerAllowedConnectionSource(String allowedUrl)
```

Comment:
If we do this, I suggest something like  Map<String,Set<String>>.  Where the key is some unique string like "BartenderProvider.class"
So registerAllowedConnection(Bartender.class.getName(), Set.ofconfiguredURL)

Reply:
After any call you could do some sort of "collect" operation (remove duplicates, I think there is a stream flattener for this) and just stash the computed concatenated string.